### PR TITLE
Add config to build Auth release on GCB

### PR DESCRIPTION
**Pull request by rvkubiak**
**Release note**:
```release-note
Add a config file for running Auth release builds on Google Container Builder.
```

### CODE REVIEW

File: bin/build_auth_release.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Used for automated builds triggered on Git tags pushed matching the regex:
+#   \d+\.\d+\.\d+
+
+steps:
+- name: 'gcr.io/$PROJECT_ID/istio-builder'
+  args: [ 'bash', './bin/push-docker.sh', '-h', "gcr.io/$PROJECT_ID", '-t', "$TAG_NAME" ]
```
Comment (sebastienvas) [inline bin/build_auth_release.yaml]: It would be good to create a release.sh, but if not use gcr.io/istio-io,docker.io/istio for -h

Comment (rvkubiak) [inline bin/build_auth_release.yaml]: Hard coded the project name.  Will add docker.io once the credential work is done.

Comment (rvkubiak) [inline bin/build_auth_release.yaml]: Added release.sh script and DockerHub push.

**Comment by sebastienvas** [comment]
/lgtm 
Need one more approver
/assign @lookuptable

**Pull request merged**